More guidance about examples (#15894)
mdn/content: files/en-us/mdn/structures/code_examples/index.md

Traditional live samples are inserted into the page using the [`EmbedLiveSample`](https://github.com/mdn/yari/blob/main/kumascript/macros/EmbedLiveSample.ejs) macro. An \\{{EmbedLiveSample}} call dynamically grabs the code blocks in the same document section as itself and puts them into a document, which it then inserts into the page inside an {{htmlelement("iframe")}}. See our [Live samples guide](/en-US/docs/MDN/Structures/Live_samples) for more information.

### Formatting live samples

> If you write a live sample, you should give it a descriptive heading, then optionally a short prose description of the example, then the following subheadings, in the following order:

Reviewer: ```suggestion
If you write a live sample in the "Examples" section, provide a descriptive H3 heading for this live sample example. Ideally, write a short description of the example explaining the scenario and what you are hoping to demonstrate. Then add subsections with following H4 headings, in the order listed:
```
an attempt to make the instructions as clear as possible

> - **HTML** containing HTML code blocks
> - **CSS** containing CSS code blocks
> - **JavaScript** containing JavaScript code blocks
> - **Result** containing the `EmbedLiveSample` macro call itself.

Reviewer: ```suggestion
- HTML
- CSS
- JavaScript
- Result
```

> 

Reviewer: ```suggestion
Write the code blocks in the respective subsections listed above.
In the **Result** subsection, add the call to the [`EmbedLiveSample` macro](/en-US/docs/MDN/Structures/Live_samples#live_sample_macros). Preferably, include some more prose in this subsection to describe the result.
```

> If you're not using a particular language type (for example if you are not using JavaScript), or if you are hiding it, then you should omit the corresponding heading.

Reviewer: ```suggestion
If you're not using a particular language type (for example, if you are not using JavaScript) or if you are hiding it, then you should omit the corresponding heading.
```

For example:

````

> 

Reviewer: ```suggestion
## Examples
```
to show the relative location in the document

### Styling a paragraph

In this example we're using CSS to style paragraphs which have the `fancy` class set.

#### HTML

```html
<p>I'm not fancy.</p>

<p class="fancy">But I am!</p>
```

#### CSS

```css
p.fancy {
  color: red;
}
```

#### Result

\{{EmbedLiveSample("Styling a paragraph")}}

> Optionally, you can include some more prose here describing the result.

Reviewer: ```suggestion
Only the `<p>` element with `class="fancy"` will get styled `red`.
```

Traditional live samples are inserted into the page using the [`EmbedLiveSample`](https://github.com/mdn/yari/blob/main/kumascript/macros/EmbedLiveSample.ejs) macro. An \\{{EmbedLiveSample}} call dynamically grabs the code blocks in the same document section as itself and puts them into a document, which it then inserts into the page inside an {{htmlelement("iframe")}}. See our [Live samples guide](/en-US/docs/MDN/Structures/Live_samples) for more information.

### Formatting live samples

> If you write a live sample in the "Examples" section, provide a descriptive H3 heading for this live sample example. Ideally, write a short description of the example explaining the scenario and what you are hoping to demonstrate. Then add subsections with following H4 headings, in the order listed:

Reviewer 2: nit:

H3  (and H4) are no more obvious I think. What about:
```suggestion
If you write a live sample in the "Examples" section, provide a descriptive H3 heading (`###`) for this live sample example. Ideally, write a short description of the example explaining the scenario and what you are hoping to demonstrate. Then add subsections with following H4 headings (`####`), in the order listed:
```

Reviewer: True, we should be conveying markdown styling over the HTML tags
